src/driver/completion.rs

```diff
@@ -24,7 +24,7 @@ impl Completion {
         }
     }
 
-    pub(crate) fn set_waker(&mut self, waker: Waker) {
+    pub(crate) fn set_waker(&self, waker: Waker) {
```

Comment:
found this bug lurking. should not be getting mutable references to the completion.